math/vecops/test/vecops_rvec.cxx

```diff
@@ -1212,6 +1212,26 @@ TEST(VecOps, DeltaPhi)
    auto dphi4 = DeltaPhi(v4, v3);
    auto r4 = -1.f * r3;
    CheckEqual(dphi4, r4);
+
+   // Checks that calling with different argument types works and yields the
+   // expected return types and values
+   static_assert(std::is_same_v<decltype(DeltaPhi(0.f, 0.0)), double>,
```

Comment:
Could you use the gtest macro EXPECT_TRUE to uniform this smart check to the rest of the code?
For example
```cpp
EXPECT_TRUE(std::is_same_v<decltype(DeltaPhi(0.f, 0.0)), double>) << "DeltaPhi should return double if one of the arguments is double";
```
This applies to other places in the code.